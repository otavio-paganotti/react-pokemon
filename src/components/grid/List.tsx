import Card from '@/components/ui/Card';
import { PokemonItem } from '@/types';

const GenericList = <T extends PokemonItem>({ items }: { items: T[] }) => {
  return (
    <>
      <div className="grid grid-cols-1 md:grid-cols-4 gap-4">
        {items && items.map((item, index) => <Card key={index} item={item} />)}
      </div>
    </>
  );
};

export default GenericList;
