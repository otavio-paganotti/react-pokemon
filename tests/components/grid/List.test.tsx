import React from 'react';
import renderer from 'react-test-renderer';
import List from '@/components/grid/List';

const toJson = (component: renderer.ReactTestRenderer) => {
  const result = component.toJSON();
  expect(result).toBeDefined();
  expect(result).not.toBeInstanceOf(Array);
  return result as renderer.ReactTestRendererJSON;
};

test('should render correctly the List component', () => {
  const component = renderer.create(<List />);

  let tree = toJson(component);
  expect(tree).toMatchSnapshot();
});
