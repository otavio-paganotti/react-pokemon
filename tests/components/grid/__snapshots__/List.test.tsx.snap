// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`should render correctly the List component 1`] = `"Here we have a list!"`;
